### PR TITLE
v.to.rast: avoid zero categories metadata when using use=cat

### LINKED ISSUE #2739: [Bug] v.to.rast writing 0 categories in metedata when use=cat
**Describe the bug**
When rasterizing with `use=cat` and then running r.info, you get (see Number of Categories):
```
...
 |   Type of Map:  raster               Number of Categories: 0               |
 |   Data Type:    CELL                 Semantic label: (none)                |
 |   Rows:         226                                                        |
 |   Columns:      220                                                        |
 |   Total Cells:  49720                                                      |
 |        Projection: Lambert Conformal Conic                                 |
 |            N:     226800    S:     220020   Res:    30                     |
 |            E:     639030    W:     632430   Res:    30                     |
 |   Range of data:    min = 696  max = 1466 
...
```
This is confusing, you would expect the number of categories to be > 0.
This was originally reported by a user.

**Additional context**
AFAIU, this used to work until https://github.com/OSGeo/grass/commit/8a1cd940ebb3574e60ca5ee57371eb4368d4fa86, which disabled the category writing due to performance reasons. So I am not sure how to address this.

### PR DESCRIPTION
**Pull request by Abhi-d-gr8**
Fixes #2739

While working with `v.to.rast`, I noticed that rasterizing with `use=cat` results in `r.info` reporting `Number of Categories: 0`, even though the raster clearly contains category values. This is confusing for users and looks like a regression.
The issue seems to come from an earlier change where writing individual category rules was disabled for performance reasons. In the `use=cat` case without labels, this means no category metadata ends up being written at all.

This PR restores meaningful category metadata by using`Rast_set_cats_fmt()` to store a category *format* instead of enumerating all categories. This keeps the fast path intact and avoids the performance cost of writing per-category rules, while ensuring `r.info` no longer reports zero categories.

I also added a small regression test to make sure `use=cat` produces non-zero category metadata going forward.

### CODE REVIEW

**Comment by wenzeslaus** [comment]
Given the starting point is a performance regression and unclear behavior, a reproducible benchmark and more reasoning is needed here. The test should be more specific to show that the desired behavior is indeed there.

File: vector/v.to.rast/testsuite/test_v_to_rast.py [hunk 1]
```diff
@@ -54,6 +55,28 @@ def test_use_interface(self):
...
+        module = SimpleModule("r.info", map=self.output, flags="gr")
+        self.runModule(module)
+        info = module.outputs.stdout
+
+        # r.info -gr uses shell format and prints ncats= (not "categories=")
+        ncats = None
+        for line in info.splitlines():
+            if line.startswith("ncats="):
+                ncats = int(float(line.split("=", 1)[1]))
+                break
```
Comment (wenzeslaus) [inline vector/v.to.rast/testsuite/test_v_to_rast.py:73]: Use grass.tool for new code.

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
See my previous comment.

**Comment by petrasovaa** [comment]
This does not change that r.info reports 0 categories, as your PR description suggests. It does add the dynamic labels, but as noted in the source code, they are not actually informative. I don't know the solution to this problem, but this is not it.
